jsapp/js/components/processing/transxDisplay.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import {formatTime} from 'js/utils';
+import type {Transx} from 'js/components/processing/singleProcessingStore';
+import bodyStyles from './processingBody.module.scss';
+import {AsyncLanguageDisplayLabel} from 'js/components/languages/languagesUtils';
+
+import styles from './transxDisplay.module.scss';
+
+interface TransxDisplayProps {
+  transx: Transx;
+}
+
+export default function TransxDisplay(props: TransxDisplayProps) {
+  function renderLanguageAndDate() {
```

Comment:
I've learned recently about [useCallback hook](https://react.dev/reference/react/useCallback) that is useful for functions defined inside a functional component. I don't think is crucial here, but wanted to make sure you know about it. It basically wraps `renderLanguageAndDate`, so it's not re-defined with every render (as it currenly happens)

Reply:
Immediately asked if using this without dependencies makes sense and found this:
https://stackoverflow.com/questions/55026139/whats-the-difference-between-usecallback-with-an-empty-array-as-inputs-and-u

makes sense in hindsight but just wanted to note this